src/services/guild.service.ts

```diff
@@ -1,13 +1,46 @@
 import { ClientService } from './client.service';
-import { Guild } from 'discord.js';
+import { Maybe } from '../common/types';
+import { Guild, GuildChannel, Role } from 'discord.js';
+import Constants from '../common/constants';
 
 export class GuildService {
   private _guild: Guild;
+  private roleCache: Record<string, Maybe<Role>> = {
+    [Constants.Roles.Unacknowledged]: undefined,
+    [Constants.Roles.Unverifed]: undefined,
+  };
+
+  private channelCache: Record<string, Maybe<GuildChannel>> = {
+    [Constants.Channels.Public.CodeOfConduct]: undefined,
+    [Constants.Channels.Bot.Verify]: undefined,
+    [Constants.Channels.Bot.Unacknowledged]: undefined,
+  };
+
   constructor(private _clientService: ClientService) {
     this._guild = this._clientService.guilds.first();
   }
 
   public get() {
     return this._guild;
   }
+
+  public getRole(roleName: string): Role {
```

Comment:
could be undefined

Reply:
we do catch when the handlers fail so it's not super bad

Reply:
It only fails if you try to search a role that doesn't exist. Currently the role it searches for are constants, so it shouldn't fail in these examples at least